geonode_mapstore_client/client/js/epics/gnresource.js

```diff
@@ -689,23 +695,35 @@ export const gnsSelectResourceEpic = (action$, store) =>
                     setResourceCompactPermissions(undefined)
                 );
             }
-            const user = userSelector(store.getState());
+            const state = store.getState();
+            const user = userSelector(state);
+            const { resource_type: resourceType, pk, subtype } = selectedResource;
+            const { resourceObservable } = resourceType === ResourceTypes.DATASET && resourceTypes[resourceType] || {};
+
             return Observable.defer(() => Promise.all([
-                getResourceByTypeAndByPk(selectedResource?.resource_type, selectedResource?.pk, selectedResource?.subtype),
+                getResourceByTypeAndByPk(resourceType, pk, subtype),
                 user
-                    ? getCompactPermissionsByPk(selectedResource?.pk)
+                    ? getCompactPermissionsByPk(pk)
                         .then((compactPermissions) => compactPermissions)
                         .catch(() => null)
                     : Promise.resolve(null)
             ]))
                 .switchMap(([resource, compactPermissions]) => {
-                    return Observable.of(
-                        setResource({
-                            ...resource,
-                            /* store information related to detail */
-                            '@ms-detail': true
-                        }),
-                        ...(compactPermissions ? [setResourceCompactPermissions(compactPermissions)] : [])
+                    return Observable.concat(
+                        Observable.of(
+                            setResourceType(resourceType),
+                            setResource({
+                                ...resource,
+                                /* store information related to detail */
+                                '@ms-detail': true
+                            }),
+                            ...(compactPermissions ? [setResourceCompactPermissions(compactPermissions)] : [])
+                        ),
+                        ...(resourceObservable ? [resourceObservable(pk, {
+                            page: "catalogue",
+                            isSamePreviousResource: true,
+                            resourceData: resource
+                        })] : [])
```

Comment:
Why do we need the resourceObservable ?
I would avoid this if possible, could you describe why this is needed?

Reply:
This is required to retrieve the mapConfiguration for the geolimit. Since it’s only needed on the Catalogue page under the Permissions tab in the Resource Detail view, I’ve utilized the existing logic that fetches the config including the dataset’s resource layer.

Reply:
ok, if it's only for geolimits I propose to use similar approach as we did for [Thumbnail](https://github.com/GeoNode/geonode-mapstore-client/blob/geonode-home/geonode_mapstore_client/client/js/plugins/ResourceDetails/components/DetailsThumbnail.jsx#L174-L177). Inside the GeoLimits component we can recreate the map with the resource and resourceToLayers function.
We should be able to remove resourceObservable and related logic with this change.